rust_src/src/floatfns.rs

```diff
@@ -307,76 +276,71 @@ pub fn copysign(x1: EmacsDouble, x2: EmacsDouble) -> EmacsDouble {
 /// The function returns the cons cell (SGNFCAND . EXP).
 /// If X is zero, both parts (SGNFCAND and EXP) are zero.
 #[lisp_fn]
-pub fn frexp(x: EmacsDouble) -> (LispObject, libc::c_int) {
-    let (significand, exponent) = libm::frexp(x);
-    (LispObject::from_float(significand), exponent)
+pub fn frexp(x: LispNumber) -> (EmacsDouble, libc::c_int) {
+    let (significand, exponent) = libm::frexp(x.to_float());
```

Comment:
return frexp straightaway
```suggestion
    libm::frexp(x.to_float())
```